zh_cn/news/_posts/2021-12-25-ruby-3-1-0-released.md

```diff
@@ -0,0 +1,248 @@
+
+基于此技术，YJIT 既可以加速启动时间，又可以提升多数现实世界中的软件性能：对 railsbench 提升22%，对 liquid-render 提升39%。
+
+YJIT 目前还是一个试验性特性，默认处于关闭状态。如果您希望使用YJIT，可以在命令行参数中加入 `--yjit` 来启用。同时，此功能目前仅在类 Unix 的 x86-64 平台中。
+
+* <https://bugs.ruby-lang.org/issues/18229>
+* <https://shopify.engineering/yjit-just-in-time-compiler-cruby>
+* <https://www.youtube.com/watch?v=PBVLf3yfMs8>
+
+## debug gem: 新调试器
+
+提供了完全重写的新调试器 [debug.gem](https://github.com/ruby/debug) 。debug.gem 有以下特点：
+
+* 改善了调试器性能（调试器不会降低应用的速度）。
+* 支持远程调试。
+* 支持富调试器前端（目前支持VS Code 和 Chrome 浏览器）。
+* 支持 多进程/多线程 调试。
+* REPL 色彩支持。
+* 以及其他有用的特性，例如 记录/重放，跟踪 等。
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.jpg">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.mp4" type="video/mp4">
+</video>
+
+Ruby 包含了 lib/debug.rb，但它没有得到良好的维护，并且在性能和特性中存在问题。debug.gem 完全替代了 lib/debug.rb。
+
+## error_highlight: 在调用栈中更好颗粒度的错误定位
+
+引入了内置的 gem error_highlight。它提供了在调用栈中更好颗粒度的错误定位：
+
+```
+$ ruby test.rb
+test.rb:1:in `<main>': undefined method `time' for 1:Integer (NoMethodError)
+
+1.time {}
+ ^^^^^
+Did you mean?  times
+```
+
+目前，仅支持 `NameError`。
+
+此 gem 默认启用。您可以通过在命令行参数中设置 `--disable-error_highlight` 来禁用。您可以参考 [error_highlight 代码库](https://github.com/ruby/error_highlight) 来了解更多详细信息。
+
+## IRB 自动补全与展示文档
+
+IRB 现在有了自动补全功能，您只需输入代码，候选补全窗口就会出现。您可以使用 Tab 和 Shift+Tab 在候选内容中进行选择。
+
+如果您安装了文档，那么当您选择了候选补全后，文档窗口将在候选补全窗口旁出现，提供对应的文档内容。您可以通过 Alt+d 来阅读完整文档。
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.png">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.mp4" type="video/mp4">
+</video>
+
+## 其它值得注意的新特性
+
+### 语言
+
+* Hash字面量和关键词参数中，值可以省略。 [[Feature #14579]](https://bugs.ruby-lang.org/issues/14579)
+  * `{x:, y:}` 等价于 `{x: x, y: y}`.
+  * `foo(x:, y:)` 等价于  `foo(x: x, y: y)`.
+
+* 在模式匹配中，pin 操作符 (`^`) 现在可以接受表达式。 [[Feature #17411]](https://bugs.ruby-lang.org/issues/17411)
+
+  ```ruby
+  Prime.each_cons(2).lazy.find_all{_1 in [n, ^(n + 2)]}.take(3).to_a
+  #=> [[3, 5], [5, 7], [11, 13]]
+  ```
+
+* 在单行模式匹配中，括号可以省略。 [[Feature #16182]](https://bugs.ruby-lang.org/issues/16182)
+
+  ```ruby
+  [0, 1] => _, x
+  {y: 2} => y:
+  x #=> 1
+  y #=> 2
+  ```
+
+### RBS
+
+RBS 是用于描述 Ruby 程序结构的语言。详情可参考 [RBS 代码库](https://github.com/ruby/rbs)。
+
+自 Ruby 3.0.0 以来的更新：
+
+* 泛型参数可以设定继承于哪个类型。 ([PR](https://github.com/ruby/rbs/pull/844))
+* 支持泛型参数别名。 ([PR](https://github.com/ruby/rbs/pull/823))
+* 引入 `rbs collection` 来管理 gems 中的 RBS。  ([文档](https://github.com/ruby/rbs/blob/master/docs/collection.md))
+* 新增/更新 了许多内建标准库的签名。
+* 包含了许多错误修正和性能提升。
+
+详情可参考 [RBS 变更日志](https://github.com/ruby/rbs/blob/master/CHANGELOG.md)。
+
+### TypeProf
+
+TypeProf 是一个 Ruby 分析器。可以从无类型标注的 Ruby 代码中生成 RBS 的原型。详情可参考 [文档](https://github.com/ruby/typeprof/blob/master/doc/doc.md)。
+
+自 Ruby 3.0.0 以来主要的更新是一个试验型 IDE 支持 “TypeProf for IDE”。
```

Comment:
Should it be `一个试验性 IDE`?

Reply:
English version: The main updates since Ruby 3.0.0 is an experimental IDE support called “TypeProf for IDE”.
Actually, it's a support tool for IDE to invoke. So, I change the translation to `一个试验型的 IDE 支持功能`